pkg/cmd/cmd_push.go

```diff
@@ -127,14 +111,51 @@ func pushTarPackage(ociUrl, localTarPath string, settings *settings.Settings) er
 		return err
 	}
 
-	// 2. Generate the OCI options from oci url and the version of current kcl package.
+	// 2. push the package
+	return pushPackage(ociUrl, kclPkg, settings)
+}
+
+// pushPackage will push the kcl package to the oci registry.
+// 1. pushPackage will package the current kcl package into default tar path.
+// 2. If the oci url is not specified, generate the default oci url from the current package.
+// 3. Generate the OCI options from oci url and the version of current kcl package.
+// 4. Push the package to the oci registry.
+func pushPackage(ociUrl string, kclPkg *pkg.KclPkg, settings *settings.Settings) error {
+	// 1. Package the current kcl package into default tar path.
+	tarPath, err := kclPkg.PackageCurrentPkgPath()
+	if err != nil {
+		return err
+	}
+
+	// clean the tar path.
+	defer func() {
+		if kclPkg != nil && utils.DirExists(tarPath) {
+			err = os.RemoveAll(tarPath)
+			if err != nil {
+				err = errors.InternalBug
+			}
+		}
+	}()
+
+	// 2. If the oci url is not specified, generate the default oci url from the current package.
+	if len(ociUrl) == 0 {
+		ociUrl, err = genDefaultOciUrlForKclPkg(kclPkg)
+		if err != nil || len(ociUrl) == 0 {
+			reporter.Report("kpm: failed to generate default oci url for current package.")
+			reporter.Report("kpm: run 'kpm push help' for more information.")
+			return errors.FailedPushToOci
+		}
+	}
+
+	// 3. Generate the OCI options from oci url and the version of current kcl package.
 	ociOpts, err := opt.ParseOciOptionFromOciUrl(ociUrl, kclPkg.GetPkgTag())
 	if err != nil {
 		return err
 	}
 
-	// 3. Push it.
-	err = oci.Push(localTarPath, ociOpts.Reg, ociOpts.Repo, ociOpts.Tag, settings)
+	reporter.Report("kpm: package '" + kclPkg.GetPkgName() + "' will be pushed.")
+	// 4. Push it.
+	err = oci.Push(tarPath, ociOpts.Reg, ociOpts.Repo, ociOpts.Tag, settings)
```

Comment:
After `push` successfully, there is no prompt. You can refer to `docker push` for optimization.

Reply:
fixed.